tests/tests.js

```diff
@@ -841,10 +841,28 @@ const namedGroupFixtures = [
 		'expectedGroups': [
 			['name', 1]
 		]
+	},
+	{
+		'pattern': '(?:(?<a>x)|(?<a>y))\\k<a>',
+		'flags': '',
+		'expected': '(?:(x)|(y))\\1\\2',
+		'expectedGroups': [
+			['a', 1],
+			['a', 2]
+		]
+	},
+	{
+		'pattern': '(?:(?<a>x)\\k<a>|(?<a>y)\\k<a>)',
+		'flags': '',
+		'expected': '(?:(x)\\1|(y)\\1\\2)',
```

Comment:
This is slightly sub-optimal (it could be `(?:(x)\\1|(y)\\2)`), but the size improvement is probably not worth the implementation complexity.

Reply:
Ack